bindings/node.js/kzg.cxx

```diff
@@ -124,46 +151,70 @@ Napi::Value FreeTrustedSetup(const Napi::CallbackInfo& info) {
   return env.Undefined();
 }
 
-// blobToKzgCommitment: (blob: Blob, setupHandle: SetupHandle) => KZGCommitment;
+/**
+ * Convert a blob to a KZG commitment.
+ *
+ * @param {Blob} blob - The blob representing the polynomial to be committed to
+ *
+ * @returns {KZGCommitment} -  The resulting commitment
+ *
+ * @throws {TypeError} - for invalid arguments or failure of the native library
+ */
 Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
-  auto env = info.Env();
-
+  Napi::Env env = info.Env();
   size_t argument_count = info.Length();
   size_t expected_argument_count = 2;
   if (argument_count != expected_argument_count) {
     return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
-
-  Blob *blob = (Blob *)extract_byte_array_from_param(info, 0, "blob");
-  if (env.IsExceptionPending()) {
+  Blob *blob = get_blob(env, info[0]);
+  if (blob == nullptr) {
     return env.Null();
   }
-
+  if (!info[1].IsExternal()) {
+     Napi::TypeError::New(env, "Must pass setupHandle as the last function argument").ThrowAsJavaScriptException();
+  }
```

Comment:
I'm not totally sure what this `IsExternal` thing is doing but maybe we forgot it in `freeTrustedSetup()`?

Perhaps we can just put this boilerplate code into a `get_kzg_settings()` function that does:
```
if (!info.isExternal()) {
  blabla
}
auto kzg_settings = info[1].As<Napi::External<KZGSettings>>().Data();
```
?

Reply:
IIUC this is a thing that will not be needed anymore if we move KZGSettings to be owned by C, right?

Reply:
Yes, exactly.  If we store the pointer to in C there will be no need to wrap it as External [here](https://github.com/ethereum/c-kzg-4844/blob/7c0bd867d5748e0420c130c91dc5d5033ebb7db0/bindings/node.js/kzg.cxx#L108) and then check every time it gets passed back in because it will be a strongly typed C object stored like [this](https://github.com/matthewkeil/c-kzg-4844/blob/99997c2cd692f2859fb69d188f48369d177bea60/bindings/node.js/kzg.cxx#L27)

`External` values in node are values that are not generated and managed by node.  Ie they are external to v8.  They allow native objects to get passed into JS and then passed back to native code (like the setup handle).

The `IsExternal` check is validating that the `napi_value` is an external value.   In JS there is no such thing as "data types" per-say.  That is what TypeScript was built for.  Instead all variables hold a "javascript value" and that value can be anything (string/number/function/etc).  All of the `Is*` functions allow for reading the meta on the value to know what actual c type the value represents.  All of the `As<Napi::*>()` functions unwrap those values and allow access to the typed JS value.

If a number value is passed into that parameter slot and `As<Napi::External>()` is called on a `Napi::Number` it will throw.